src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -645,7 +645,7 @@ private void ProjectFinished(object sender, ProjectFinishedEventArgs e)
                             string urlString = url.ToString();
                             if (Uri.TryCreate(urlString, UriKind.Absolute, out Uri? uri))
                             {
-                                urlString = uri.AbsoluteUri;
+                                urlString = uri.ToString();
```

Comment:
I think this deserves a comment: why does the other API mangle it? Should we not use this `Uri.TryCreate` at all?

Reply:
I agree, I'll add the comment.
Here is what i found 
 https://github.com/Azure/azure-sdk-for-net/issues/12102

Reply:
done